### DISCUSSION TITLE
magick identify can't parse iTXt?

### DISCUSSION THREAD

**Discussion by vt-idiot** [ImageMagick/ImageMagick#6000] [Help]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

Windows

### Description

Version: ImageMagick 7.1.0-57 Q16-HDRI x64 eadf378:20221230

Try `magick identify -verbose` on a PNG with iTXt chunks. It will not print them. Trying to view an iTXt chunk with a known name returns an error. Can it just not read them? It works fine when they are Latin/ANSI tEXt.

**Comment by dlemstra** [comment]
Might be a good idea to share a sample file?

**Comment by vt-idiot** [comment]
Oops, good point @dlemstra 

I am not sure if github strips PNG metadata when uploaded, guess I'll find out.

I manually edited the tag for demonstration, but these results match those of files that "originally" had Unicode iTXt tags instead of ANSI tEXt.

## All tags are "tEXt", no Unicode:
![kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink s-421623618](https://user-images.githubusercontent.com/81622808/214908507-b3622ff5-0df7-4a61-b0d5-48b1c030160a.png)
```cmd
C:\test\gh>magick identify -format %[Title]\n%[Description]\n%[Software]\n%[Source]\n%[Comment] -verbose "kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink s-421623618.png"
AI generated image
masterpiece, best quality, kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink hairbow, slurping, bowl, close-up, holding chopsticks
NovelAI
Stable Diffusion 1D44365E
{"steps": 50, "sampler": "k_euler", "seed": 421623618, "strength": 0.6, "noise": 0.1, "scale": 12.0, "uc": "lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry"}
```
[output.tEXt.txt](https://github.com/ImageMagick/ImageMagick/files/10511971/output.tEXt.txt)
> png:text: 5 tEXt/zTXt/iTXt chunks were found
## `Description` tag is "iTXt" with Unicode characters, remaining tags are "tEXt", no Unicode:
![kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink s-421623618 iTXt](https://user-images.githubusercontent.com/81622808/214908822-ddb4f55d-56c1-459a-b419-b347eba95b37.png)
```cmd
C:\test\gh>magick identify -format %[Title]\n%[Description]\n%[Software]\n%[Source]\n%[Comment] -verbose "kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink s-421623618.iTXt.png"
identify: unknown image property "%[Description]" @ warning/property.c/InterpretImageProperties/4236.
AI generated image

NovelAI
Stable Diffusion 1D44365E
{"steps": 50, "sampler": "k_euler", "seed": 421623618, "strength": 0.6, "noise": 0.1, "scale": 12.0, "uc": "lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry"}
C:\test\gh>magick identify -verbose "kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink s-421623618.iTXt.png" > output.txt
```
[output.txt](https://github.com/ImageMagick/ImageMagick/files/10511954/output.txt)
> png:text: 4 tEXt/zTXt/iTXt chunks were found

`magick identify` can't even see the tag anymore. But it's clearly still in there:
![image](https://user-images.githubusercontent.com/81622808/214909779-64f37d53-8264-45ff-9829-14d8472875ee.png)

I thought it might be because the Language tag is not set. Note: changing the language tag on hundreds of files to make this work is not something I plan on doing, but it didn't work anyways, even after only leaving Japanese characters in the iTXt tag, and setting the language to `ja-JP`
![キズナアイ拉麺 iTXt ja-JP](https://user-images.githubusercontent.com/81622808/214911745-fbdfb647-71b3-4ce2-a646-e1e6c1153d9b.png)
![image](https://user-images.githubusercontent.com/81622808/214911419-c9c3001e-5899-4392-af08-75bf2ec1da77.png)
```cmd
C:\test\gh>magick identify -verbose "キズナアイ拉麺.iTXt.ja-JP.png" > output.キズナアイ拉麺.ja-JP.txt
```
[output.キズナアイ拉麺.ja-JP.txt](https://github.com/ImageMagick/ImageMagick/files/10512053/output.ja-JP.txt)
At least it can still handle the filenames when the codepage allows for it.

**Comment by fmw42** [comment]
I am not sure I follow all this, but I get the following on IM 7.1.0.58 on Mac OSX Monterey

```
magick identify -verbose 214908507-b3622ff5-0df7-4a61-b0d5-48b1c030160a.png

Image:
  Filename: 214908507-b3622ff5-0df7-4a61-b0d5-48b1c030160a.png
  Permissions: rw-r--r--
  Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: DirectClass
  Geometry: 768x768+0+0
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColor
  Endianness: Undefined
  Depth: 8-bit
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
  Channel statistics:
    Pixels: 589824
    Red:
      min: 0  (0)
      max: 255 (1)
      mean: 168.999 (0.662741)
      median: 180 (0.705882)
      standard deviation: 75.0567 (0.29434)
      kurtosis: -1.28333
      skewness: -0.346918
      entropy: 0.910044
    Green:
      min: 4  (0.0156863)
      max: 255 (1)
      mean: 138.34 (0.542509)
      median: 137 (0.537255)
      standard deviation: 74.5158 (0.292219)
      kurtosis: -1.32046
      skewness: 0.109738
      entropy: 0.963531
    Blue:
      min: 1  (0.00392157)
      max: 255 (1)
      mean: 127.19 (0.498784)
      median: 110 (0.431373)
      standard deviation: 77.7587 (0.304936)
      kurtosis: -1.38372
      skewness: 0.236487
      entropy: 0.96198
  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 144.843 (0.568011)
      median: 142.333 (0.55817)
      standard deviation: 75.7771 (0.297165)
      kurtosis: -1.40958
      skewness: -0.00146088
      entropy: 0.945185
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: black
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 768x768+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Zip
  Orientation: Undefined
  Properties:
    Comment: {"steps": 50, "sampler": "k_euler", "seed": 421623618, "strength": 0.6, "noise": 0.1, "scale": 12.0, "uc": "lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry"}
    date:create: 2023-01-26T18:36:12+00:00
    date:modify: 2023-01-26T18:36:10+00:00
    date:timestamp: 2023-01-26T18:36:38+00:00
    Description: masterpiece, best quality, kizuna ai eating ramen like a normal person, leaning down, leaning forward, pink hairbow, slurping, bowl, close-up, holding chopsticks
    png:IHDR.bit-depth-orig: 8
    png:IHDR.bit_depth: 8
    png:IHDR.color-type-orig: 2
    png:IHDR.color_type: 2 (Truecolor)
    png:IHDR.interlace_method: 0 (Not interlaced)
    png:IHDR.width,height: 768, 768
    png:text: 5 tEXt/zTXt/iTXt chunks were found
    signature: 5327b67f2aaf2f03c633beb006c6245e18439681e6e30230d32ec3484dc35fbc
    Software: NovelAI
    Source: Stable Diffusion 1D44365E
    Title: AI generated image
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 698125B
  Number pixels: 589824
  Pixel cache type: Memory
  Pixels per second: 37.8302MP
  User time: 0.020u
  Elapsed time: 0:01.015
  Version: ImageMagick 7.1.0-58 Q16-HDRI aarch64 20802 https://imagemagick.org

```

<br>
which shows (but does not print): 
 
<br>

` png:text: 5 tEXt/zTXt/iTXt chunks were found`

**Reply by vt-idiot** [reply]
> on Mac OSX Monterey

Then it's likely Windows specific. Still doesn't help me if it won't show the iTXt chunk.

I made what I was trying to do using exiftool...which can't actually read Unicode from the command prompt, no matter what, so I had to kludge writing the filename into an `ASCII version of the filename.txt` and pass that as an argument to it. Problem solved for my use case, in a very very ugly way, but identify still ought to be able to read them...

**Reply by fmw42** [reply]
Seems a reasonable request to me.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.
